### PR TITLE
fix: Use requests to fetch event and speaker images

### LINKED ISSUE #6618: Image resizing is not working
https://github.com/fossasia/open-event-server/blob/9bb9f68305c261713ee486f3840ebcf55dc70d23/app/api/helpers/files.py#L80

urlib is throwing 403 Forbidden error for almost all images. 

Please use requests and verify image resizing works. Also, create a management command which finds all events which have no resized images and resizes them

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #6618 

#### Short description of what this resolves:
Resolves image-resizing task which was throwing error 403

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by codedsun** [comment]
@iamareebjamal The test fails and gives Corrupt/Invalid images for few

https://github.com/python-pillow/Pillow/issues/3287

For now, the image resizing is working with local images as well as postman

**Comment by codedsun** [comment]
```[2019-12-06 00:49:38,953: ERROR/ForkPoolWorker-2] Task resize.user.images[9d4b41a5-1249-4c0a-ac2b-5a3ed3866990] raised unexpected: OSError('Corrupt/Invalid Image')
Traceback (most recent call last):
  File "/Users/codedsun/Documents/FOSS/open-event-server/app/api/helpers/files.py", line 85, in create_save_resized_image
    im = Image.open(image_file)
  File "/usr/local/lib/python3.7/site-packages/PIL/Image.py", line 2818, in open
    raise IOError("cannot identify image file %r" % (filename if filename else fp))
OSError: cannot identify image file <_io.BytesIO object at 0x11266bc50>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/celery/app/trace.py", line 385, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/flask_celeryext/app.py", line 101, in __call__
    res = Task.__call__(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/celery/app/trace.py", line 648, in __protected_call__
    return self.run(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/sentry_sdk/integrations/celery.py", line 158, in _inner
    reraise(*exc_info)
  File "/usr/local/lib/python3.7/site-packages/sentry_sdk/_compat.py", line 57, in reraise
    raise value
  File "/usr/local/lib/python3.7/site-packages/sentry_sdk/integrations/celery.py", line 153, in _inner
    return f(*args, **kwargs)
  File "/Users/codedsun/Documents/FOSS/open-event-server/app/api/helpers/tasks.py", line 147, in resize_user_images_task
    uploaded_images = create_save_image_sizes(original_image_url, 'speaker-image', user.id)
  File "/Users/codedsun/Documents/FOSS/open-event-server/app/api/helpers/files.py", line 154, in create_save_image_sizes
    'original_image_url': create_save_resized_image(image_file, 0, 0, 0, original_upload_path, resize=False),
  File "/Users/codedsun/Documents/FOSS/open-event-server/app/api/helpers/files.py", line 87, in create_save_resized_image
    raise IOError("Corrupt/Invalid Image")
OSError: Corrupt/Invalid Image
```

Log

**Comment by iamareebjamal** [comment]
That's because `r.raw.read()` is empty.

It's simply `Image.open(io.BytesIO(response.content))`

File: app/api/helpers/files.py [hunk 1]
```diff
@@ -77,7 +78,7 @@ def create_save_resized_image(image_file, basewidth=None, maintain_aspect=None,
     if not image_file:
         return None
     filename = '{filename}.{ext}'.format(filename=get_file_name(), ext=ext)
-    data = urllib.request.urlopen(image_file).read()
+    data = requests.get(image_file, stream=True).content
```
Comment (iamareebjamal) [inline app/api/helpers/files.py]: Why stream=True?

**Comment by codedsun** [comment]
>That's because r.raw.read() is empty

Why the raw.read() is empty and how the response.content is valid?

File: app/api/helpers/files.py (hunk 1 above)
Comment (codedsun) [reply app/api/helpers/files.py]: Was previously using raw, have updated.

**Comment by iamareebjamal** [comment]
dredd tests are broken. Fix them

**Comment by codedsun** [comment]
@iamareebjamal Shall i skip that api tests?

**Comment by codedsun** [comment]
@iamareebjamal Check now!

**Comment by iamareebjamal** [comment]
Please don't skip the transactions, just delete the entire dredd tests as well. Why test anything, right?

**Comment by codedsun** [comment]
Sorry, I didn't mean that.

**Comment by iamareebjamal** [comment]
It's simple example.com/example.png does not exist

**Comment by iamareebjamal** [comment]
```
INFO:werkzeug:127.0.0.1 - - [05/Dec/2019 21:07:46] "PATCH /v1/users/2 HTTP/1.1" 500 -
ERROR:werkzeug:Error on request:
Traceback (most recent call last):
  File "/home/circleci/code/venv/lib/python3.7/site-packages/werkzeug/serving.py", line 304, in run_wsgi
    execute(self.server.app)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/werkzeug/serving.py", line 292, in execute
    application_iter = app(environ, start_response)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 70, in sentry_patched_wsgi_app
    environ, start_response
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 120, in __call__
    reraise(*_capture_exception(hub))
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/_compat.py", line 57, in reraise
    raise value
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 116, in __call__
    _sentry_start_response, start_response, span
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 69, in <lambda>
    return SentryWsgiMiddleware(lambda *a, **kw: old_app(self, *a, **kw))(
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/circleci/code/app/__init__.py", line 77, in __call__
    return self.app(environ, start_response)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/app.py", line 1866, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/circleci/code/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/api.py", line 133, in decorated
    return self.check_permissions(view, view_args, view_kwargs, *args, **kwargs)
  File "/home/circleci/code/app/api/helpers/permission_manager.py", line 488, in permission_manager
    return permissions[args[0]](view, view_args, view_kwargs, *args, **kwargs)
  File "/home/circleci/code/app/api/helpers/permissions.py", line 46, in decorator
    return fn(*args, **kwargs)
  File "/home/circleci/code/app/api/helpers/permission_manager.py", line 127, in is_user_itself
    return view(*view_args, **view_kwargs)
  File "/home/circleci/code/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/decorators.py", line 32, in wrapper
    return func(*args, **kwargs)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask/views.py", line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File "/home/circleci/code/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/resource.py", line 68, in dispatch_request
    response = method(*args, **kwargs)
  File "/home/circleci/code/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/decorators.py", line 56, in wrapper
    return func(*args, **kwargs)
  File "/home/circleci/code/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/resource.py", line 310, in patch
    self._data_layer.update_object(obj, data, kwargs)
  File "/home/circleci/code/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/data_layers/alchemy.py", line 126, in update_object
    self.before_update_object(obj, data, view_kwargs)
  File "/home/circleci/code/app/api/users.py", line 266, in before_update_object
    start_image_resizing_tasks(user, data['avatar_url'])
  File "/home/circleci/code/app/api/users.py", line 326, in start_image_resizing_tasks
    resize_user_images_task.delay(user_id, original_image_url)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/celery/app/task.py", line 427, in delay
    return self.apply_async(args, kwargs)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/integrations/celery.py", line 97, in apply_async
    return f(*args, **kwargs)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/celery/app/task.py", line 552, in apply_async
    link=link, link_error=link_error, **options)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/celery/app/task.py", line 774, in apply
    ret = tracer(task_id, args, kwargs, request)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/integrations/celery.py", line 135, in _inner
    return f(*args, **kwargs)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/celery/app/trace.py", line 399, in trace_task
    I, R, state, retval = on_error(task_request, exc, uuid)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/celery/app/trace.py", line 385, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/flask_celeryext/app.py", line 101, in __call__
    res = Task.__call__(self, *args, **kwargs)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/celery/app/task.py", line 394, in __call__
    return self.run(*args, **kwargs)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/integrations/celery.py", line 158, in _inner
    reraise(*exc_info)
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/_compat.py", line 57, in reraise
    raise value
  File "/home/circleci/code/venv/lib/python3.7/site-packages/sentry_sdk/integrations/celery.py", line 153, in _inner
    return f(*args, **kwargs)
  File "/home/circleci/code/app/api/helpers/tasks.py", line 147, in resize_user_images_task
    uploaded_images = create_save_image_sizes(original_image_url, 'speaker-image', user.id)
  File "/home/circleci/code/app/api/helpers/files.py", line 155, in create_save_image_sizes
    small_upload_path),
  File "/home/circleci/code/app/api/helpers/files.py", line 96, in create_save_resized_image
    width_percent = (basewidth / float(img.size[0]))
TypeError: unsupported operand type(s) for /: 'NoneType' and 'float'
```

**Comment by codedsun** [comment]
@iamareebjamal What shall be done for this issue
```
  width_percent = (basewidth / float(img.size[0]))
TypeError: unsupported operand type(s) for /: 'NoneType' and 'float'
```
I can understand that basewidth is becoming none, but when we are calling the function we are passing the basewidth for every image size, is it somwhere we are passing none to the function means image basewidth is coming to be None?

**Comment by iamareebjamal** [comment]
That, you have to debug as to why it is becoming None

**Comment by codedsun** [comment]
@iamareebjamal Okay debugging, taking time for it.

File: app/api/helpers/files.py [hunk 2]
```diff
@@ -77,6 +78,7 @@ def create_save_resized_image(image_file, basewidth=None, maintain_aspect=None,
     if not image_file:
         return None
     filename = '{filename}.{ext}'.format(filename=get_file_name(), ext=ext)
+    # data = requests.get(image_file).content
```
Comment (iamareebjamal) [inline app/api/helpers/files.py]: Why? Debug things locally. Don't push intermediate changes

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4625055&bid=15506282)

File: app/api/helpers/files.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 import urllib.parse
 import urllib.request
 import uuid
+import requests
```
Comment (niranjan94) [inline app/api/helpers/files.py]: Codacy Issue found: [Unused import requests](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4625055)

Comment (niranjan94) [inline app/api/helpers/files.py]: Codacy Issue found: ['requests' imported but unused](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4625055)

**Comment by codedsun** [comment]
@iamareebjamal I logged locally all the basewidth sizes, they are coming to be appropriate sizes and working fine for both the API's. I will take more time to debug, I don't know what is getting wrong.

**Comment by codedsun** [comment]
@iamareebjamal I installed request module using pip, do we need to put that in requirements also?

**Comment by iamareebjamal** [comment]
It's already in requirements

**Comment by iamareebjamal** [comment]
So you're saying that you cannot reproduce it locally?

**Comment by codedsun** [comment]
> So you're saying that you cannot reproduce it locally?

for dredd test or api hit?


For API Hit via postman - I cant' reproduce it locally, all are working fine for me

**Comment by iamareebjamal** [comment]
> for dredd test or api hit?

Doesn't matter. Same thing

**Comment by iamareebjamal** [comment]
That's because you have image sizes saved locally while they are not when running the tests

**Comment by codedsun** [comment]
Then what should be done for this case?

**Comment by iamareebjamal** [comment]
Sizes should be added

**Comment by codedsun** [comment]
>That's because you have image sizes saved locally while they are not present in DB when running the tests

How come they are not present in the DB? Also, I checked for api.eventyay.com, they are present but I didn't checked for dev envt. Can you provide me the link of dev envt api, also for the size, do  I need to send the query or how they are added Please tell me and I will be happy to proceed.

**Comment by iamareebjamal** [comment]
Tests are not run on dev environment

**Comment by iamareebjamal** [comment]
Checks dredd hooks file

**Comment by codedsun** [comment]
How do I add the sizes now?

**Comment by iamareebjamal** [comment]
Because nothing is present in the DB unless you add it explicitly

**Comment by codedsun** [comment]
@iamareebjamal Added the sizes but still failing ;(

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/open-event-server/blob/6b1f5db/app/factories/image_size.py#L37

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/open-event-server/blob/9bb9f68305c261713ee486f3840ebcf55dc70d23/app/api/helpers/files.py#L129-L132

This means that it should work without image sizes being saved in the DB. It's some other issue

**Comment by codedsun** [comment]
> https://github.com/fossasia/open-event-server/blob/9bb9f68305c261713ee486f3840ebcf55dc70d23/app/api/helpers/files.py#L129-L132
> 
> This means that it should work without image sizes being saved in the DB. It's some other issue

But all the sizes which are used in the function are not defined. then what's the use ??


Also for 

> https://github.com/fossasia/open-event-server/blob/6b1f5db/app/factories/image_size.py#L37


For the function we are using only these sizes which are present in the factory, how does `thumbnail_size_quality` breaks it?

```
       thumbnail_aspect = icon_aspect = small_aspect = True
        thumbnail_basewidth = thumbnail_height_size = image_sizes.thumbnail_size_width_height
        icon_basewidth = icon_height_size = image_sizes.icon_size_width_height
        small_basewidth = small_height_size = image_sizes.small_size_width_height
```

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/open-event-server/blob/9bb9f68305c261713ee486f3840ebcf55dc70d23/app/api/helpers/files.py#L140-L142

thumbnail_size_width_height, icon_size_width_height, and small_size_width_height is used and no value is set for these above. Wow

**Comment by codedsun** [comment]
@iamareebjamal Yes, that's what I was pointing, then what's the use of it here.

**Comment by iamareebjamal** [comment]
Because `type` is not set correctly in factory, hence it is never fetched, exception is raised and default ImageSizes is returned which has no thumbnail_size_width_height, icon_size_width_height, small_size_width_height

**Comment by codedsun** [comment]
> Because `type` is not set correctly in factory, hence it is never fetched, exception is raised and default ImageSizes is returned which has no thumbnail_size_width_height, icon_size_width_height, small_size_width_height

Okay what shall be the value of type?

**Comment by iamareebjamal** [comment]
> then what's the use of it here

That's the wrong question to ask. The right response is, now I know why things are failing. Fixing it right away

**Comment by iamareebjamal** [comment]
What's present in the factory

**Comment by codedsun** [comment]
> What's present in the factory

https://github.com/fossasia/open-event-server/blob/6b1f5db/app/factories/image_size.py#L37

Asking for the value you pointed here, for the `type` will change to `event-image` and `speaker-image`

**Comment by iamareebjamal** [comment]
Also, open an issue to add default values of image sizes in the schema and DB

**Comment by iamareebjamal** [comment]
Factories won't be used now. Remove them

**Comment by codedsun** [comment]
@iamareebjamal 
https://github.com/fossasia/open-event-server/blob/6d098cb7548a82f85129e4b817bef13170ec2dec/populate_db.py#L87-L104


What shall be the values of other params which are not present here for event-size image, speaker-image? I can't see the values here.

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/open-event-server/pull/6639#issuecomment-562668600

**Comment by codedsun** [comment]
Some absurd and confusing values which are not complete are present in the factory,

https://github.com/fossasia/open-event-server/blob/6d098cb7548a82f85129e4b817bef13170ec2dec/app/factories/image_size.py#L12-L26

Also,
the fallback is present here 
https://github.com/fossasia/open-event-server/blob/6d098cb7548a82f85129e4b817bef13170ec2dec/app/api/helpers/files.py#L162-L171

Shall i use these values written in else case ?

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/helpers/files.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4625055&bid=15506282)

**Comment by codedsun** [comment]
@iamareebjamal Pheew!!!. This has passed all the test, please check.
Thanks for your patience and bearing it up with me. This has taken long and was very good learning for me.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/helpers/files.py  1
         

Clones added
============
- app/api/helpers/files.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4625055&bid=15506282)

File: app/api/helpers/files.py [hunk 4]
```diff
@@ -129,8 +130,17 @@ def create_save_image_sizes(image_file, image_sizes_type, unique_identifier=None
...
+            image_sizes = ImageSizes(image_sizes_type, icon_size_width_height=35, icon_size_quality=80,
+                                     small_size_width_height=50, small_size_quality=80,
+                                     thumbnail_size_width_height=500, thumbnail_quality=80)
+        else:
+            image_sizes = ImageSizes(image_sizes_type, full_width=1300,
+                                     full_height=500, full_aspect=True, full_quality=80,
+                                     icon_width=75, icon_height=30, icon_aspect=True,
+                                     icon_quality=80, thumbnail_width=500, thumbnail_height=200,
+                                     thumbnail_aspect=True, thumbnail_quality=80, logo_width=500,
+                                     logo_height=200)
```
Comment (iamareebjamal) [inline app/api/helpers/files.py]: No need for condition, add all values unconditionally

Comment (codedsun) [reply app/api/helpers/files.py]: The values are different for speaker-type and event-type

File: docs/api/blueprint/user/users.apib [hunk 5]
```diff
@@ -213,7 +213,7 @@ Create a new user using an email, password and an optional name.
                   "twitter-url": "http://twitter.com/twitter",
                   "instagram-url": "http://instagram.com/instagram",
                   "google-plus-url": "http://plus.google.com/plus.google",
-                  "original-image-url": "https://cdn.pixabay.com/photo/2013/11/23/16/25/birds-216412_1280.jpg"
```
Comment (iamareebjamal) [inline docs/api/blueprint/user/users.apib]: No need to change this

Comment (codedsun) [reply docs/api/blueprint/user/users.apib]: ok

File: docs/api/blueprint/user/users.apib [hunk 6]
```diff
@@ -559,9 +559,9 @@ Authorized user should be same as user in request body or must be admin.
                   "twitter-url": "http://twitter.com/twitter",
                   "instagram-url": "http://instagram.com/instagram",
                   "google-plus-url": "http://plus.google.com/plus.google",
-                  "thumbnail-image-url": "http://example.com/example.png",
-                  "small-image-url": "http://example.com/example.png",
-                  "icon-image-url": "http://example.com/example.png"
+                  "thumbnail-image-url": "https://image.flaticon.com/icons/png/64/126/126486.png",
+                  "small-image-url": "https://image.flaticon.com/icons/png/64/126/126486.png",
+                  "icon-image-url": "https://image.flaticon.com/icons/png/64/126/126486.png"
```
Comment (iamareebjamal) [inline docs/api/blueprint/user/users.apib]: These are read only and should be removed

File: docs/api/blueprint/user/users.apib [hunk 7]
```diff
@@ -204,7 +204,7 @@ Create a new user using an email, password and an optional name.
                 {
                   "email": "email@example.com",
                   "password": "password",
-                  "avatar_url": "http://example.com/example.png",
+                  "avatar_url": "https://image.flaticon.com/icons/png/64/126/126486.png",
```
Comment (iamareebjamal) [inline docs/api/blueprint/user/users.apib]: Use https://avatars2.githubusercontent.com/u/1583873

File: docs/api/blueprint/user/users.apib (hunk 6 above)
Comment (codedsun) [reply docs/api/blueprint/user/users.apib]: remove these fields or revert the change? Removing from the documentation response will be like the documentation is incomplete? Your call, tell me what

File: docs/api/blueprint/user/users.apib [hunk 8]
```diff
@@ -550,7 +550,7 @@ Authorized user should be same as user in request body or must be admin.
               "data": {
                 "attributes": {
                   "password": "password",
-                  "avatar_url": "http://example.com/example.png",
+                  "avatar_url": "https://image.flaticon.com/icons/png/64/126/126486.png",
```
Comment (iamareebjamal) [inline docs/api/blueprint/user/users.apib]: Use this https://avatars2.githubusercontent.com/u/1583873

File: docs/api/blueprint/user/users.apib (hunk 7 above)
Comment (codedsun) [reply docs/api/blueprint/user/users.apib]: Ohh, mario 👍  :D ok

File: docs/api/blueprint/user/users.apib (hunk 6 above)
Comment (iamareebjamal) [reply docs/api/blueprint/user/users.apib]: No, they are discarded. So, the documentation will be complete

Comment (codedsun) [reply docs/api/blueprint/user/users.apib]: @iamareebjamal

File: app/api/helpers/files.py (hunk 4 above)
Comment (iamareebjamal) [reply app/api/helpers/files.py]: No, they are mutually exclusive

File: docs/api/blueprint/user/users.apib (hunk 6 above)
Comment (codedsun) [reply docs/api/blueprint/user/users.apib]: @iamareebjamal Test are failing because the request becomes incomplete

Comment (iamareebjamal) [reply docs/api/blueprint/user/users.apib]: No, they are failing because it's invalid JSON

Comment (codedsun) [reply docs/api/blueprint/user/users.apib]: reverting this change to the previous one.

Comment (iamareebjamal) [reply docs/api/blueprint/user/users.apib]: You can't have trailing commas in JSON

File: docs/api/blueprint/user/users.apib [hunk 9]
```diff
@@ -559,9 +559,6 @@ Authorized user should be same as user in request body or must be admin.
                   "twitter-url": "http://twitter.com/twitter",
                   "instagram-url": "http://instagram.com/instagram",
                   "google-plus-url": "http://plus.google.com/plus.google",
```
Comment (iamareebjamal) [inline docs/api/blueprint/user/users.apib]: JSON can't have trailing commas

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- app/api/helpers/files.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4625055&bid=15506282)

File: docs/api/blueprint/user/users.apib (hunk 9 above)
Comment (codedsun) [reply docs/api/blueprint/user/users.apib]: That was a blunder, apologies.

**Comment by codedsun** [comment]
Check now!

**Comment by iamareebjamal** [comment]
Great work. Now, work on creating a management task which resizes the images of events and speakers which were not done previously

**Comment by codedsun** [comment]
This wasn't great as I used to do, it had lot of blunders from my side. Thnkyou for the help and efforts 👍 May be my bad health made it more worse but yes I did even in this case ^_^

>Now, work on creating a management task which resizes the images of events and speakers which were not done previously

New Issue for this?

**Comment by iamareebjamal** [comment]
No, follow up to this PR as it was a part of this issue only

**Comment by codedsun** [comment]
Can you give me any example of management-task written in the repo.

>No, follow up to this PR as it was a part of this issue only

Okay, will send a new PR with the same issue tag.

**Comment by iamareebjamal** [comment]
manage.py

**Comment by codedsun** [comment]
@iamareebjamal I have created the management command but directly invoking the function of create_resized image, the urllib gives connection refused error. Do I need to invoke the same celery task?

**Comment by iamareebjamal** [comment]
> urllib gives connection refused error

You replaced urllib with requests yourself

**Comment by codedsun** [comment]
@iamareebjamal Yes, sorry I fixed that. Also do i need to create seperate management commands for events and speakers or one? Thanks

**Comment by codedsun** [comment]
> > urllib gives connection refused error
> 
> You replaced urllib with requests yourself

Yes and facing and fixing the dredd tests was fantastic tasks :P

**Comment by iamareebjamal** [comment]
One

**Pull request merged**